### ISSUE TITLE
The log item of "Batch size per GPU" may not work as expected

### ISSUE THREAD

**Issue by AlfredQin** [Deci-AI/super-gradients#1483]
### 🐛 Describe the bug

The call of log function in the sg_trainer is shown as below, https://github.com/Deci-AI/super-gradients/blob/15c0fe1390cc307f0a1a98e576b964f494ad1e83/src/super_gradients/training/sg_trainer/sg_trainer.py#L1385 the log function gets batch size information by using len(inputs). The inputs is expected to be an input tensor with shape B, C, H, W so that the len(inputs) can get the batch size. However, if the inputs is a dict, e.g. like: ``` {"image": torch.Tensor(B, C, H, W), "other_items": others}```, the len(inputs) always outputs the length of the dict no matter what exactly the batch size is. 

```        
log_main_training_params(
            multi_gpu=device_config.multi_gpu,
            num_gpus=get_world_size(),
            batch_size=len(inputs),
            batch_accumulate=self.batch_accumulate,
            train_dataset_length=len(self.train_loader.dataset),
            train_dataloader_len=len(self.train_loader),
        )
```

### Versions

NA

**Comment by BloodAxe** [comment]
Currently the SG is not designed to work with dict (It will probably crash) so the incorrect batch size is a least problem here. When we will be adding dict support, we will address this for sure.
As of now I don't see any actions to be made here. However thanks for bringing this up and future-testing our code ;)
